core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkStandardLogger.java

```diff
@@ -49,18 +50,25 @@ private SdkStandardLogger() {
      * Log the response status code and request ID
      */
     public static void logRequestId(SdkHttpResponse response) {
-        String placeholder = "not available";
-        String requestId = String.format("Request ID: %s, Extended Request ID: %s",
-                                         SdkHttpUtils.firstMatchingHeaderFromCollection(response.headers(),
-                                                                                        X_AMZN_REQUEST_ID_HEADERS)
-                                                     .orElse(placeholder),
-                                         response.firstMatchingHeader(X_AMZ_ID_2_HEADER)
-                                                 .orElse(placeholder));
-        Supplier<String> logStatement = () -> String.format("Received %s response: %s, %s",
-                                                            response.isSuccessful() ? "successful" : "failed",
-                                                            response.statusCode(),
-                                                            requestId);
+        Supplier<String> logStatement = () -> constructLog(response);
         REQUEST_ID_LOGGER.debug(logStatement);
         REQUEST_LOGGER.debug(logStatement);
     }
+
+    private static String constructLog(SdkHttpResponse response) {
+        String placeholder = "not available";
+        String prefix = "Received " + (response.isSuccessful() ? "successful" : "failed") +
+                        " response: ";
+        StringJoiner details = new StringJoiner(", ", prefix, "");
+        details.add("Status Code: " + response.statusCode());
+        String requestId = SdkHttpUtils.firstMatchingHeaderFromCollection(response.headers(),
+                                                       X_AMZN_REQUEST_ID_HEADERS)
+                    .orElse(placeholder);
+
+        details.add("Request ID: " + requestId);
+
+        response.firstMatchingHeader(X_AMZ_ID_2_HEADER).ifPresent(extendedRequestId ->
+                                                                      details.add("Extended Request ID: " + extendedRequestId));
+        return details.toString();
```

Comment:
While we're worrying about efficiency, should we just use StringBuilder? This feels hard to read with StringJoiner.

Reply:
Yup, makes sense